xlators/features/shard/src/shard.c

```diff
@@ -4865,10 +4867,14 @@ shard_opendir(call_frame_t *frame, xlator_t *this, loc_t *loc, fd_t *fd,
     STACK_WIND(frame, shard_opendir_cbk, FIRST_CHILD(this),
                FIRST_CHILD(this)->fops->opendir, loc, fd, xdata);
 
+    return 0;
```

Comment:
`xdata` must be unref'd before returning.

Reply:
My bad again